doc/src/guides/solving/solve-matrix-equation.md

```diff
@@ -0,0 +1,352 @@
+
+```py
+>>> from sympy import symbols, Matrix, simplify
+>>> c, d, e = symbols("c, d, e")
+>>> A = Matrix([[c,d], [1, -e]])
+>>> b = Matrix([2, 0])
+>>> b
+    ⎡2⎤
+    ⎢ ⎥
+    ⎣0⎦
+>>> b2 = Matrix([4, 0])
+>>> b2
+    ⎡4⎤
+    ⎢ ⎥
+    ⎣0⎦
+>>> inv = A.inv()
+>>> inv
+    ⎡   e        d   ⎤
+    ⎢───────  ───────⎥
+    ⎢c⋅e + d  c⋅e + d⎥
+    ⎢                ⎥
+    ⎢   1       -c   ⎥
+    ⎢───────  ───────⎥
+    ⎣c⋅e + d  c⋅e + d⎦
+>>> solution = inv * b # Solves Ax = b for x
+>>> solution
+    ⎡  2⋅e  ⎤
+    ⎢───────⎥
+    ⎢c⋅e + d⎥
+    ⎢       ⎥
+    ⎢   2   ⎥
+    ⎢───────⎥
+    ⎣c⋅e + d⎦
+>>> simplify(A * solution) # Demonstrate that solution is correct
+    ⎡2⎤
+    ⎢ ⎥
+    ⎣0⎦
+>>> solution2 = inv * b2 # Solves Ax = b2 for x
+>>> solution2
+    ⎡  4⋅e  ⎤
+    ⎢───────⎥
+    ⎢c⋅e + d⎥
+    ⎢       ⎥
+    ⎢   4   ⎥
+    ⎢───────⎥
+    ⎣c⋅e + d⎦
+>>> simplify(A * solution2) # Demonstrate that solution2 is correct
+    ⎡4⎤
+    ⎢ ⎥
+    ⎣0⎦
+```
+
+Determining the inverse of a large symbolic matrix may not be computationally
+tractable.
+
+### Work With Symbolic Matrices
+
+The computational complexity of manipulating symbolic matrices can increase
+rapidly with matrix size. For example, the number of terms in the determinant of
+a symbolic matrix increases with the factorial of the matrix dimension. As a
+result, the maximum dimensionality of matrices that can be solved is more
+limited than for numerical matrices. For example, the determinant of this 4x4
+symbolic matrix has 24 terms with four elements in each term:
+
+```py
+>>> from sympy import MatrixSymbol
+>>> A = MatrixSymbol('A', 4, 4).as_explicit()
+>>> A
+⎡A₀₀  A₀₁  A₀₂  A₀₃⎤
+⎢                  ⎥
+⎢A₁₀  A₁₁  A₁₂  A₁₃⎥
+⎢                  ⎥
+⎢A₂₀  A₂₁  A₂₂  A₂₃⎥
+⎢                  ⎥
+⎣A₃₀  A₃₁  A₃₂  A₃₃⎦
+>>> A.det()
+    A₀₀⋅A₁₁⋅A₂₂⋅A₃₃ - A₀₀⋅A₁₁⋅A₂₃⋅A₃₂ - A₀₀⋅A₁₂⋅A₂₁⋅A₃₃ + A₀₀⋅A₁₂⋅A₂₃⋅A₃₁ + A₀₀⋅A₁
+    ₃⋅A₂₁⋅A₃₂ - A₀₀⋅A₁₃⋅A₂₂⋅A₃₁ - A₀₁⋅A₁₀⋅A₂₂⋅A₃₃ + A₀₁⋅A₁₀⋅A₂₃⋅A₃₂ + A₀₁⋅A₁₂⋅A₂₀⋅
+    A₃₃ - A₀₁⋅A₁₂⋅A₂₃⋅A₃₀ - A₀₁⋅A₁₃⋅A₂₀⋅A₃₂ + A₀₁⋅A₁₃⋅A₂₂⋅A₃₀ + A₀₂⋅A₁₀⋅A₂₁⋅A₃₃ - 
+    A₀₂⋅A₁₀⋅A₂₃⋅A₃₁ - A₀₂⋅A₁₁⋅A₂₀⋅A₃₃ + A₀₂⋅A₁₁⋅A₂₃⋅A₃₀ + A₀₂⋅A₁₃⋅A₂₀⋅A₃₁ - A₀₂⋅A₁
+    ₃⋅A₂₁⋅A₃₀ - A₀₃⋅A₁₀⋅A₂₁⋅A₃₂ + A₀₃⋅A₁₀⋅A₂₂⋅A₃₁ + A₀₃⋅A₁₁⋅A₂₀⋅A₃₂ - A₀₃⋅A₁₁⋅A₂₂⋅
+    A₃₀ - A₀₃⋅A₁₂⋅A₂₀⋅A₃₁ + A₀₃⋅A₁₂⋅A₂₁⋅A₃₀
+```
+
+and solving a matrix equation of it takes about a minute, whereas the analogous
+3x3 matrix takes less than one second. The more unrelated, symbolic entries in a
+matrix, the more likely it is to be slow to manipulate. This example, finding a
+general solution to a matrix where all elements are independent symbols, is the
+extreme case and thus the slowest for a matrix of its size.
+
+### Speed up Solving Matrix Equations
+Here are some suggestions:
+- If matrix elements are zero, ensure that they are recognized as zero. You can
+  do this by either making them zero or by applying
+  [assumptions](assumptions_module).
+- Selecting a solve method suited to the properties of the matrix, for example
```

Comment:
@moorepants Are there specific solving functions, or flags for solving functions, that we should recommend to users who have those kinds of matrix (hermitian, symmetric, triangular, etc.)?